In [86]:
import sqlite3
from sqlite3 import Error

 
def create_connection(db_file):
    """ create a database connection to the SQLite database
        specified by the db_file
    :param db_file: database file
    :return: Connection object or None
    """
    conn = None
    try:
        conn = sqlite3.connect(db_file)
        print(sqlite3.version)
    except Error as e:
        print(e)
 
    return conn

 
def select_all_tasks(conn):
    """
    Query all rows in the tasks table
    :param conn: the Connection object
    :return:
    """
    cur = conn.cursor()
    
    query1 = """
    select 
sum(case when B.memid > 0 and F.facid = 0 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Tennis Court 1',
sum(case when B.memid > 0 and F.facid = 1 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Tennis Court 2',
sum(case when B.memid > 0 and F.facid = 2 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Badminton Court',
sum(case when B.memid > 0 and F.facid = 3 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Table Tennis',
sum(case when B.memid > 0 and F.facid = 4 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Massage Room 1',
sum(case when B.memid > 0 and F.facid = 5 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Massage Room 2',
sum(case when B.memid > 0 and F.facid = 6 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Squash Court',
sum(case when B.memid > 0 and F.facid = 7 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Snooker Table',
sum(case when B.memid > 0 and F.facid = 8 and B.starttime LIKE '2012-09%' then 1 else 0 end) as 'Pool Table'
from Bookings as B join Facilities as F using(facid) 
"""
    cur.execute(query1)
 
    rows = cur.fetchall()
 
    for row in rows:
        print(row)


def main():
    database = "sqlite_db_pythonsqlite.db"
 
    # create a database connection
    conn = create_connection(database)
    with conn: 
        print("2. Query all tasks")
        select_all_tasks(conn)
 
 
if __name__ == '__main__':
    main()

2.6.0
2. Query all tasks
(132, 126, 161, 194, 191, 14, 87, 199, 408)


In [40]:
#Total revenue per facility by member, Code Below for values, 3 month total revenues 
import numpy as np
F = [1540,1380,0, 0, 4167.9,267.3,682.5,0, 0]

#Total revnue per facility by guest
G = [2500,2825,604.5, 90,16640,6720, 4287.5,115,265]

#Maintenance for three months for each facility
M = [200,200,50,10,3000,3000,80,15,15]

Revenue = np.subtract(np.add(F,G),np.multiply(3,M))
print(Revenue) #Revenue less than 1000 are facid: 2,3,5,7,8 Made new profit table

"""Solution: ('Badminton Court', 454.5)
('Table Tennis', 60)
('Massage Room 2', -2012.7)
('Snooker Table', 70)
('Pool Table', 220)"""


[ 3440.   3605.    454.5    60.  11807.9 -2012.7  4730.     70.    220. ]


Question 10 Code: Repeated below query for when memid = 0 to find guest revenue
Select facid,memid,
SUM(CASE
	WHEN facid = 0 and memid > 0 then membercost
    ELSE 0
END) as Tennis_1,
SUM(CASE
	WHEN facid = 1 and memid > 0 then membercost
    ELSE 0
END) as Tennis_2,
SUM(CASE
	WHEN facid = 2 and memid > 0 then membercost
    ELSE 0
END) as B_Court,
SUM(CASE
	WHEN facid = 3 and memid > 0 then membercost
    ELSE 0
END) as Table_T
SUM(CASE
	WHEN facid = 4 and memid > 0 then membercost
    ELSE 0
END) as Massage_1,
SUM(CASE
	WHEN facid = 5 and memid > 0 then membercost
    ELSE 0
END) as Massage_2,
SUM(CASE
	WHEN facid = 6 and memid > 0 then membercost
    ELSE 0
END) as S_Court,
SUM(CASE
	WHEN facid = 7 and memid > 0 then membercost
    ELSE 0
END) as S_Table,
SUM(CASE
	WHEN facid = 8 and memid > 0 then membercost
    ELSE 0
END) as P_Table
FROM Bookings
JOIN Facilities using(facid)

Select name,Total_Profit.Revenues
FROM Facilities 
Join Total_Profit using(facid)
where Total_Profit.Revenues < 1000;

Question 11 Code: select M1.memid,concat(M1.firstname,' ',M1.surname),M1.recommendedby,concat(M2.firstname,' ',M2.surname) as Rec_Person
from Members M1, Members M2 where M1.recommendedby = M2.memid and M1.recommendedby > 0
order by M2.surname;

Solution: (24, 'Ramnaresh', 'Sarwin', '15', 'Florence', 'Bader')
(22, 'Joan', 'Coplin', '16', 'Timothy', 'Baker')
(20, 'Matthew', 'Genting', '5', 'Gerald', 'Butters')
(16, 'Timothy', 'Baker', '13', 'Jemima', 'Farrell')
(17, 'David', 'Pinker', '13', 'Jemima', 'Farrell')
(27, 'Henrietta', 'Rumney', '20', 'Matthew', 'Genting')
(26, 'Douglas', 'Jones', '11', 'David', 'Jones')
(7, 'Nancy', 'Dare', '4', 'Janice', 'Joplette')
(11, 'David', 'Jones', '4', 'Janice', 'Joplette')
(35, 'John', 'Hunt', '30', 'Millicent', 'Purview')
(8, 'Tim', 'Boothe', '3', 'Tim', 'Rownam')
(4, 'Janice', 'Joplette', '1', 'Darren', 'Smith')
(5, 'Gerald', 'Butters', '1', 'Darren', 'Smith')
(10, 'Charles', 'Owen', '1', 'Darren', 'Smith')
(14, 'Jack', 'Smith', '1', 'Darren', 'Smith')
(21, 'Anna', 'Mackenzie', '1', 'Darren', 'Smith')
(29, 'Henry', 'Worthington-Smyth', '2', 'Tracy', 'Smith')
(30, 'Millicent', 'Purview', '2', 'Tracy', 'Smith')
(36, 'Erica', 'Crumpet', '2', 'Tracy', 'Smith')
(12, 'Anne', 'Baker', '9', 'Ponder', 'Stibbons')
(15, 'Florence', 'Bader', '9', 'Ponder', 'Stibbons')
(9, 'Ponder', 'Stibbons', '6', 'Burton', 'Tracy')

Question 12 Code: 
select 
sum(case when B.memid > 0 and F.facid = 0 then 1 else 0 end) as 'Tennis Court 1',
sum(case when B.memid > 0 and F.facid = 1 then 1 else 0 end) as 'Tennis Court 2',
sum(case when B.memid > 0 and F.facid = 2 then 1 else 0 end) as 'Badminton Court',
sum(case when B.memid > 0 and F.facid = 3 then 1 else 0 end) as 'Table Tennis',
sum(case when B.memid > 0 and F.facid = 4 then 1 else 0 end) as 'Massage Room 1',
sum(case when B.memid > 0 and F.facid = 5 then 1 else 0 end) as 'Massage Room 2',
sum(case when B.memid > 0 and F.facid = 6 then 1 else 0 end) as 'Squash Court',
sum(case when B.memid > 0 and F.facid = 7 then 1 else 0 end) as 'Snooker Table',
sum(case when B.memid > 0 and F.facid = 8 then 1 else 0 end) as 'Pool Table'
from Bookings as B join Facilities as F using(facid) 


Solution: (308, 276, 344, 385, 421, 27, 195, 421, 783)

Question 13 Code: Repeat for every month like replace 07 with 08 then 09
    
select 
sum(case when B.memid > 0 and F.facid = 0 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Tennis Court 1',
sum(case when B.memid > 0 and F.facid = 1 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Tennis Court 2',
sum(case when B.memid > 0 and F.facid = 2 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Badminton Court',
sum(case when B.memid > 0 and F.facid = 3 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Table Tennis',
sum(case when B.memid > 0 and F.facid = 4 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Massage Room 1',
sum(case when B.memid > 0 and F.facid = 5 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Massage Room 2',
sum(case when B.memid > 0 and F.facid = 6 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Squash Court',
sum(case when B.memid > 0 and F.facid = 7 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Snooker Table',
sum(case when B.memid > 0 and F.facid = 8 and B.starttime LIKE '2012-07%' then 1 else 0 end) as 'Pool Table'
from Bookings as B join Facilities as F using(facid) 

Solution: July-->(65, 41, 51, 48, 77, 4, 23, 68, 103)
    August-->(111, 109, 132, 143, 153, 9, 85, 154, 272)
    September-->(132, 126, 161, 194, 191, 14, 87, 199, 408)